### ISSUE TITLE
Cannot process waveform for file "xxx.m4a"

### ISSUE THREAD

**Issue by Moonbase59** [AzuraCast/AzuraCast#7241]
### Installation Method

Docker Installation

### AzuraCast Release Channel

Rolling Release Channel

### Current AzuraCast Version

Rolling Release #1fec896 (2024-06-25 0:57)

### What happened?

When trying to use the Visual Cue Editor on a M4A file, an error message is shown briefly:

```
Cannot process waveform for file "xxx.m4a": Can’t generate json format output from Unknown file format
```

![AzuraCast VCE M4A](https://github.com/AzuraCast/AzuraCast/assets/3706922/984dd985-fd89-45c1-b657-9a5de333b25e)

A few seconds later, the waveform and correct cue/fade points _are_ shown, though.

**Comment by netbladenl** [comment]
Can confirm, same issue here. After a few seconds the waveform shows up

**Comment by BusterNeece** [comment]
@Moonbase59 This was somewhat working as intended, but I went ahead and did a bit of code work that I had been putting off previously.

Basically, up until now, we called PHP looking for either a cached JSON waveform file in the filesystem or to generate one using the onboard `audiowaveform` binary we have in the Docker image. It's not a huge problem if the script can't return a waveform, as often the JS library can generate its own (as you see in this process). However, we weren't then saving the waveform that the JS library generated back to the filesystem to both avoid future errors and improve performance.

As of the latest Rolling Release, we're now doing that, so that client-side waveform generation will only happen once, and it'll save that back to the filesystem just as if it was done with `audiowaveform`, so future requests will load the waveform editor almost instantly. If the PHP-side script fails for any reason, it'll just silently ignore that error and let the JS side handle it, so you won't see that error any more either.
